conda/base/context.py

```diff
@@ -1168,6 +1174,25 @@ def cpu_flags(self):
         info = _get_cpu_info()
         return info["flags"]
 
+    @memoizedproperty
+    @unique_sequence_map(unique_key="backend")
+    def reporters(self) -> tuple[Mapping[str, str]]:
+        """
+        Determine the value of reporters based on other settings and the ``self._reporters``
+        value itself.
+        """
+        if not self._reporters:
+            return (
+                {
+                    "backend": "json" if self.json else "stdlib",
+                    "output": "stdout",
+                    "verbosity": self.verbosity,
+                    "quiet": self.quiet,
+                },
+            )
```

Comment:
I assumed `json` would become an own reporter and not tied to stdout? E.g. `--json` is a shortcut for `--reporter=json`?

Also, why do we need the `reporters` setting? Wouldn't that list of reporters come from plugins that provide a reporter implementation?

Conda would ship with two out of the box:

```suggestion
        if not self._reporters:
            return (
                {
                    "backend": "plaintext",
                    "output": "stdout",
                    "verbosity": self.verbosity,
                    "quiet": self.quiet,
                },
                {
                    "backend": "json",
                    "output": "stdout",
                    "verbosity": self.verbosity,
                    "quiet": self.quiet,
                 },
             )
```

Reply:
@jezdez,

The reporters listed there are the currently active ones, not the available ones.

The `reporters` setting is meant to be a convenience for other parts of the code and to make it easier to link it with what has been defined in the configuration file. 

It doesn't do anything right now, but as we progress with the refactor we will be relying on `context.reporters` as the source of truth for all the output rendering that needs to occur for the running process, which could be multiple things depending on what has been configured (e.g. we could choose both `output = file` and `output = stdout` to write to both stdout and a file at the same time).

Please let me know if this makes sense and whether you would like to discuss it more.

Reply:
> Please let me know if this makes sense and whether you would like to discuss it more

Ah, gotcha, I misunderstood the purpose, thanks for elaborating!